controllers/ovncontroller_controller.go

```diff
@@ -601,35 +600,10 @@ func (r *OVNControllerReconciler) reconcileNormal(ctx context.Context, instance
 
 	sbCluster, err := ovnv1.GetDBClusterByType(ctx, helper, instance.Namespace, map[string]string{}, ovnv1.SBDBType)
 	if err != nil {
-		Log.Info("No SB OVNDBCluster defined, deleting external ConfigMap")
-		cleanupConfigMapErr := r.deleteExternalConfigMaps(ctx, helper, instance)
-		if cleanupConfigMapErr != nil {
-			Log.Error(cleanupConfigMapErr, "Failed to delete external ConfigMap")
-			return ctrl.Result{}, cleanupConfigMapErr
-		}
+		Log.Info("No SB OVNDBCluster defined. Exiting reconcile.")
 		return ctrl.Result{}, nil
```

Comment:
is it correct to return "success" status when 1) there is OVNController but 2) there is no SB db? I thought that we are making sure that OVNController doesn't exist in openstack-operator; and if it does exist, then either openstack-operator is wrong, or SB down - also a problem. Should we then raise an error?

Reply:
I think so, I didn't want to modify it as I thought it was outside of the scope of the fix (as in the current code if this happens it won't return an error). But I agree that this is a configuration (no SB enabled/spawned) that we don't support.

My only question here is, could it happen that we reach this point because the OVN resources (ONVController,OVNDBCluster,OVNNorthd) were just created, and OVNController was the first one to be reconciled and when this point is reached it's just that OVNDBCluster was "slower" and it's not yet created? If we return an error, OVNController will get reconciled again later? Or it will remain on error state?

Reply:
I've thought about it for a bit, and I think now that the current logic is fine. It will not do anything until SB is created. When it's created, the reconcile for OVNController will trigger again and it will have a chance to handle it. Until then, from the perspective of the OVNController, there is no work to do. It's a success.

About your ordering concern, I think the above addresses it: when SB is up, OVNController will have another change to reconcile.